src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt

```diff
@@ -145,18 +145,25 @@ object RsCommonCompletionProvider : CompletionProvider<CompletionParameters>() {
         }
 
         for (elementName in CompletionUtil.sortMatching(result.prefixMatcher, keys)) {
-            val candidates = AutoImportFix.getImportCandidates(importContext, elementName, elementName) {
-                !(it.item is RsMod || it.item is RsModDeclItem || it.item.parent is RsMembers)
-            }
-
-            candidates
+            getImportCandidates(importContext, elementName)
                 .distinctBy { it.qualifiedNamedItem.item }
-                .map { candidate ->
-                    val item = candidate.qualifiedNamedItem.item
+                .mapNotNull { originalCandidate ->
+                    val item = originalCandidate.qualifiedNamedItem.item
+
+                    val (scopeName, candidate) = if (item is RsEnumVariant) {
+                        val enum = item.parentEnum
+                        val enumName = enum.name ?: return@mapNotNull null
+                        val candidate = getImportCandidates(importContext, enumName)
```

Comment:
Is it not possible to do it without calling `getImportCandidates` multiple times?